tests/lib/lte_lc/CMakeLists.txt

```diff
@@ -18,19 +18,48 @@ cmock_handle(${ZEPHYR_NRFXLIB_MODULE_DIR}/nrf_modem/include/nrf_modem_at.h
 FILE(GLOB app_sources src/*.c)
 target_sources(app PRIVATE ${app_sources})
 
+FILE(GLOB lib_module_sources ${ZEPHYR_NRF_MODULE_DIR}/lib/lte_link_control/modules/*.c)
+FILE(GLOB lib_common_sources ${ZEPHYR_NRF_MODULE_DIR}/lib/lte_link_control/common/*.c)
+
 target_sources(app
   PRIVATE
-  ${ZEPHYR_NRF_MODULE_DIR}/lib/lte_link_control/lte_lc_helpers.c
+  ${lib_module_sources}
+  ${lib_common_sources}
 )
 
 target_include_directories(app
   PRIVATE
   ${ZEPHYR_NRF_MODULE_DIR}/lib/lte_link_control/
+  ${ZEPHYR_NRF_MODULE_DIR}/lib/lte_link_control/include
   ${ZEPHYR_NRF_MODULE_DIR}/include/modem/
+  ${ZEPHYR_NRFXLIB_MODULE_DIR}/nrf_modem/include
 )
 
 target_compile_options(app
   PRIVATE
   -DCONFIG_LTE_LINK_CONTROL_LOG_LEVEL=0
   -DCONFIG_LTE_NEIGHBOR_CELLS_MAX=10
+  -DCONFIG_LTE_LC_WORKQUEUE_STACK_SIZE=512
+  -DCONFIG_LTE_LC_TAU_PRE_WARNING_TIME_MS=0
+  -DCONFIG_LTE_LC_MODEM_SLEEP_PRE_WARNING_TIME_MS=0
+  -DCONFIG_LTE_LC_MODEM_SLEEP_NOTIFICATIONS_THRESHOLD_MS=0
+  -DCONFIG_LTE_LC_TAU_PRE_WARNING_THRESHOLD_MS=0
+  -DCONFIG_LTE_MODE_PREFERENCE_VALUE=0
+  -DCONFIG_LTE_PSM_REQ_RAT="00100001"
+  -DCONFIG_LTE_PSM_REQ_RPTAU="00000011"
+  -DCONFIG_LTE_NETWORK_TIMEOUT=0
+  -DCONFIG_LTE_EDRX_REQ_VALUE_LTE_M="1001"
+  -DCONFIG_LTE_EDRX_REQ_VALUE_NBIOT="1001"
+  -DCONFIG_LTE_PTW_VALUE_LTE_M=""
+  -DCONFIG_LTE_PTW_VALUE_NBIOT=""
+  -DCONFIG_LTE_LC_CONEVAL=1
+  -DCONFIG_LTE_LC_EDRX=1
+  -DCONFIG_LTE_LC_NCELLMEAS=1
+  -DCONFIG_LTE_LC_PERIODICSEARCHCONF=1
+  -DCONFIG_LTE_LC_PSM=1
+  -DCONFIG_LTE_LC_RAI=1
+  -DCONFIG_LTE_LC_REDMOB=1
+  -DCONFIG_LTE_LC_XFACTORYRESET=1
+  -DCONFIG_LTE_LC_XMODEMSLEEP=1
+  -DCONFIG_LTE_LC_TAU_PRE_WARNING=1
```

Comment:
why is this not coming / controlled through Kconfig ?

Same comment applies to other test CMakeLists.txt files.

Reply:
Because this test cannot use the link controller directly, yet it needs to be able to see the various types and features in `lte_lc.h` that are behind the Kconfig toggles (and these Kconfig options are visible only when enabling `CONFIG_LTE_LINK_CONTROL`). Now the alternative that was implemented before was to add an extra `#if defined(CONFIG_UNITY)` after each flagging in the header, but we discussed and decided that it would make the header unnecessarily confusing, so instead we opted for the current solution, which is a mere extension of what has been done already before.

Reply:
When fine-tuning `lte_lc` test set I'd like to mention that I have a plan to remove this test set (I've already removed some of it's test earlier) and cover all the test coverage it has in `lte_lc_api` tests which enable `CONFIG_LTE_LINK_CONTROL` and use `lte_lc` only through its public API and mocks underneath.

Reply:
tests/lib/lte_lc have been removed so resolving this comment.

Reply:
I think this sounds like something we should take a closer look at when re-introducing the tests.
Perhaps a small KConfig reorganization for lte lc can help on this.

The `unit_testing` board upstream supports Kconfig exactly to avoid the need of constructs like this, especially because they easily gets out of sync.

Feel free to do an offline ping when you want to re-introduce the tests :+1: .

Reply:
These tests are not coming back. There is another test set (lte_lc_api) which covers all those scenarios and since those tests use `CONFIG_LTE_LINK_CONTROL=y`, these Kconfig options are available to be set in `prj.conf` etc.